### PR TITLE
Adds a tgui interface for the slot machine

### PR DESCRIPTION
**Pull request by ZeWaka**
[UI] [FEAT]

## About the PR 
Adds a tgui interface for the slot machine, serving as an intermediate introduction to tgui

## Why's this needed? 
* pretty

### CODE REVIEW

**Comment by ZeWaka** [comment]
Held up due to https://github.com/goonstation/goonstation/issues/1739

File: code/obj/submachine/slots.dm [hunk 1]
```diff
@@ -6,155 +6,147 @@
...
+		tgui_not_incapacitated_state.can_use_topic(src, user)
+	)
+
+/obj/submachine/slot_machine/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if(..())
 		return
+	switch(action)
+		if ("insert_card")
+			var/obj/O = usr.equipped()
+			if (istype(O, /obj/item/card/id))
```
Comment (mordent-goonstation) [inline code/obj/submachine/slots.dm:51]: Need to check that there's no ID already inserted (never trust the front-end).

File: code/obj/submachine/slots.dm [hunk 2]
```diff
@@ -6,155 +6,147 @@
...
+			ui_interact(usr, ui)
+			SPAWN_DBG(2.5 SECONDS) // why was this at ten seconds, christ
+				money_roll()
+				src.working = 0
+				src.icon_state = "slots-off"
+
+		if("eject")
+			if(!src.scan)
+				return TRUE // jerks doing that "hide in a chute to glitch auto-update windows out" exploit caused a wall of runtime errors
+			src.scan.set_loc(get_turf(src))
```
Comment (mordent-goonstation) [inline code/obj/submachine/slots.dm:79]: I know this is parity with existing implementation, but would doing `usr.put_in_hand_or_eject(src.scan)` be better?

File: tgui/packages/tgui/interfaces/SlotMachine.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+const SlotWindow = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { scannedCard, money, plays } = data;
+
+  return (
+    <Box>
+      <NoticeBox success>
+        <marquee> Twenty credits to play! </marquee>
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/SlotMachine.js:58]: You monster.

File: tgui/packages/tgui/interfaces/SlotMachine.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        icon="dice"
+        content="Play!"
+        tooltip="Pull the lever"
+        tooltipPosition="right"
+        onClick={() => act('play')} />
+    </Box>
+  );
+};
+
+const BusyWindow = (props, context) => {
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/SlotMachine.js:87]: If you're not using them, no need to have `props` and `context` as arguments.

File: tgui/packages/tgui/interfaces/SlotMachine.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+export const SlotMachine = (props, context) => {
+  const { data } = useBackend(context);
+  const { scannedCard, busy } = data;
+  return (
+    <Window
+      title="Slot Machine"
+      width={375}
+      height={175}
+      resizable>
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/SlotMachine.js:18]: Does it need to be resizable?

File: tgui/packages/tgui/interfaces/SlotMachine.js (hunk 3 above)
Comment (ZeWaka) [reply tgui/packages/tgui/interfaces/SlotMachine.js:58]: It actually looks great!

File: tgui/packages/tgui/interfaces/SlotMachine.js (hunk 5 above)
Comment (ZeWaka) [reply tgui/packages/tgui/interfaces/SlotMachine.js:18]: I mean, why not?

Comment (mordent-goonstation) [reply tgui/packages/tgui/interfaces/SlotMachine.js:18]: I don't think the content's dynamic in size (so the set width/height is probably always what you actually want), but that's fair I guess.

File: code/obj/submachine/slots.dm [hunk 6]
```diff
@@ -6,155 +6,149 @@
...
-			if(src.scan)
-				boutput(user, "<span class='alert'>There is a card already in the slot machine.</span>")
-			else
-				boutput(user, "<span class='notice'>You insert your ID card.</span>")
-				usr.drop_item()
-				I.set_loc(src)
-				src.scan = I
-				src.updateUsrDialog()
-		else src.attack_hand(user)
+/obj/submachine/slot_machine/ui_interact(mob/user, datum/tgui/ui)
```
Comment (mordent-goonstation) [inline code/obj/submachine/slots.dm:21]: Idle thought: could we make this boilerplate into a `#define`? It's a multi-line one so maybe not, but I imagine this is going to be added a *lot* as we gradually add tgui to things.

Out of scope for this PR.

File: tgui/packages/tgui/interfaces/SlotMachine.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2020 @ZeWaka
+ * SPDX-License-Identifier: ISC
+ */
+
+import { useBackend } from '../backend';
+import { Box, Button, NoticeBox, Divider, BlockQuote, Icon } from '../components';
+import { Window } from '../layouts';
+
+export const SlotMachine = context => {
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/SlotMachine.js:10]: Double check this works. I know I said to remove unneeded arguments, but I think `context` is the second argument passed in?

File: code/obj/submachine/slots.dm (hunk 6 above)
Comment (ZeWaka) [reply code/obj/submachine/slots.dm:21]: technically yes

**Pull request merged**